src/backend/distributed/utils/citus_clauses.c

```diff
@@ -305,27 +91,52 @@ EvaluateNodeIfReferencesFunction(Node *expression, PlanState *planState)
 		case T_ArrayCoerceExpr:
 		case T_ScalarArrayOpExpr:
 		case T_RowCompareExpr:
-		case T_Param:
 		case T_RelabelType:
 		case T_CoerceToDomain:
 		{
+			if (FindNodeCheck(expression, IsVarNode))
```

Comment:
I'm not a huge fan of the style of this function (I'd prefer an explicit return variable and more `break` and a single-return and moving the `T_Const` block down here), but since you're not authoring it, just changing it, I guess this is OK.